presto-hive/src/test/java/com/facebook/presto/hive/util/TestHiveFileIterator.java

```diff
@@ -0,0 +1,223 @@
+    {
+        // set up
+        File rootDir = createTempDir();
+        String basePath = rootDir.getAbsolutePath();
+        // create 8 files in root directory - 3 pathFilter matched and 5 non matched files.
+        createFiles(basePath, 3, true);
+        createFiles(basePath, 5, false);
+        Path rootPath = new Path("file://" + basePath + File.separator);
+        PathFilter pathFilter = path -> path.getName().contains(pathFilterString);
+        HiveFileIterator hiveFileIterator = new HiveFileIterator(rootPath, listDirectoryOperation, new NamenodeStats(), IGNORED, pathFilter);
+
+        int actualCount = getFileCount(hiveFileIterator);
+        assertEquals(actualCount, 3);
+
+        // cleanup
+        deleteTestDir(rootDir);
+    }
+
+    @Test
+    public void testPathFilterWithRecursion() throws IOException
+    {
+        // set up
+        File rootDir = createTempDir();
+        String basePath = rootDir.getAbsolutePath();
+        // create 8 files in root directory - 3 pathFilter matched and 5 non matched files.
+        createFiles(basePath, 3, true);
+        createFiles(basePath, 5, false);
+        // create two directories
+        List<File> subDirs = createDirs(basePath, 2);
+        // create 5 files in dir1 - 3 pathFilter matched and 2 non matched files.
+        String dir1 = subDirs.get(0).getAbsolutePath();
+        createFiles(dir1, 3, true);
+        createFiles(dir1, 2, false);
+        // create 7 files in dir2 - 3 pathFilter matched and 4 non matched files.
+        String dir2 = subDirs.get(1).getAbsolutePath();
+        createFiles(dir2, 3, true);
+        createFiles(dir2, 4, false);
+        Path rootPath = new Path("file://" + basePath + File.separator);
+        PathFilter pathFilter = path -> path.getName().contains(pathFilterString);
+        HiveFileIterator hiveFileIterator = new HiveFileIterator(rootPath, listDirectoryOperation, new NamenodeStats(), RECURSE, pathFilter);
+
+        int actualCount = getFileCount(hiveFileIterator);
+        assertEquals(actualCount, 9);
+
+        // cleanup
+        deleteTestDir(rootDir);
+    }
+
+    private void deleteTestDir(File rootDir)
+    {
+        if (rootDir != null && rootDir.exists()) {
+            delete(rootDir);
+        }
+    }
+
+    private void delete(File file)
+    {
+        if (!file.exists()) {
+            return;
+        }
+        if (file.isDirectory()) {
+            for (File child : file.listFiles()) {
+                delete(child);
+            }
+        }
+        file.delete();
+    }
+
+    private void createFiles(String basePath, int numFiles, boolean matchPathFilter) throws IOException
+    {
+        for (int i = 0; i < numFiles; i++) {
+            new File(basePath).mkdirs();
+            String fileName;
+            if (matchPathFilter) {
+                fileName = pathFilterString + "_" + i;
+            }
+            else {
+                fileName = getRandomStringOfLen(10);
```

Comment:
does it really have to be generated randomly? Can it be simply `"__not_matched_prefix_" + i`  instead?

Reply:
fixed